analysis/topEFT/topeft.py

```diff
@@ -277,8 +289,21 @@ def process(self, events):
 
         # Weights
         genw = np.ones_like(events['MET_pt']) if isData else events['genWeight']
-        weights = coffea.analysis_tools.Weights(len(events))
-        weights.add('norm',genw if isData else (xsec/sow)*genw)
+
+        ### We need weights for: normalization, lepSF, triggerSF, pileup, btagSF...
+        weights = {}
+        for r in ['all', 'ee', 'mm', 'em', 'eee', 'mmm', 'eem', 'mme']:
+          weights[r] = coffea.analysis_tools.Weights(len(events))
+          weights[r].add('norm',genw if isData else (xsec/sow)*genw)
+
+        weights['ee'].add('lepSF_eeSS', lepSF_eeSS)
+        weights['em'].add('lepSF_emSS', lepSF_emSS)
+        weights['mm'].add('lepSF_mmSS', lepSF_mumuSS)
+        weights['eee'].add('lepSF_eee', lepSF_eee)
+        weights['mmm'].add('lepSF_mmm', lepSF_mmm)
+        weights['mme'].add('lepSF_mme', lepSF_mme)
+        weights['eem'].add('lepSF_eem', lepSF_eem)
+
```

Comment:
Maybe we should fold this into the for loop above, and just add a lepSF if r isn't all?

Reply:
Well, then we'd need to build a dict or a list with the lepSF arrays so they can be pointed to in the loop.

Reply:
Good point, we'll leave it as is.